### PR TITLE
Add hostname env integration test.

### PR DESCRIPTION
**Pull request by Random-Liu**
Fixes https://github.com/containerd/cri/issues/904.

Add hostname env integration test.
Signed-off-by: Lantao Liu <lantaol@google.com>

### CODE REVIEW

File: integration/test_utils.go [hunk 1]
```diff
@@ -126,6 +126,13 @@ func WithPodLogDirectory(dir string) PodSandboxOpts {
 	}
 }
 
+// Add pod hostname.
+func WithHostname(hostname string) PodSandboxOpts {
```
Comment (mikebrow) [inline integration/test_utils.go]: nit /s/WithHostname/WithPodHostname/

File: integration/hostname_env_test.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	runtime "k8s.io/kubernetes/pkg/kubelet/apis/cri/runtime/v1alpha2"
+)
+
+func TestHostnameEnv(t *testing.T) {
```
Comment (mikebrow) [inline integration/hostname_env_test.go]: nit /s/TestHostNameEnv/TestPodHostNameEnv/

same for name of file..

File: integration/hostname_env_test.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+			},
+			expectedHostname: "test-hostname",
+		},
+		"host network pod with hostname": {
+			opts: []PodSandboxOpts{
+				WithHostNetwork,
+			},
+			expectedHostname: hostname,
+		},
+		"host network pod without hostname": {
```
Comment (mikebrow) [inline integration/hostname_env_test.go]: /s/without/with custom/

File: integration/hostname_env_test.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+)
+
+func TestHostnameEnv(t *testing.T) {
+	hostname, err := os.Hostname()
+	require.NoError(t, err)
+	for name, test := range map[string]struct {
+		opts             []PodSandboxOpts
+		expectedHostname string
+	}{
+		"regular pod with hostname": {
```
Comment (mikebrow) [inline integration/hostname_env_test.go]: /s/with/with custom/

File: integration/hostname_env_test.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+			opts: []PodSandboxOpts{
+				WithHostname("test-hostname"),
+			},
+			expectedHostname: "test-hostname",
+		},
+		"host network pod with hostname": {
```
Comment (mikebrow) [inline integration/hostname_env_test.go]: /s/with/with default/

File: integration/test_utils.go (hunk 1 above)
Comment (Random-Liu) [reply integration/test_utils.go]: Done

File: integration/hostname_env_test.go (hunk 2 above)
Comment (Random-Liu) [reply integration/hostname_env_test.go]: Done

File: integration/hostname_env_test.go (hunk 3 above)
Comment (Random-Liu) [reply integration/hostname_env_test.go]: Done

File: integration/hostname_env_test.go (hunk 4 above)
Comment (Random-Liu) [reply integration/hostname_env_test.go]: Done

File: integration/hostname_env_test.go (hunk 5 above)
Comment (Random-Liu) [reply integration/hostname_env_test.go]: Done

**Pull request merged**